rust_src/src/windows.rs

```diff
@@ -135,3 +136,18 @@ pub fn window_minibuffer_p(window: LispObject) -> LispObject {
     };
     LispObject::from_bool(win.as_window_or_error().is_minibuffer())
 }
+
+/// Return the window which was selected when entering the minibuffer.
+/// Returns nil, if selected window is not a minibuffer window.
+#[lisp_fn]
+pub fn minibuffer_selected_window() -> LispObject {
+    let level = unsafe { minibuf_level };
+    let current_minibuf = unsafe { LispObject::from_raw(current_minibuf_window) };
+    if level > 0 && selected_window().as_window_or_error().is_minibuffer() &&
+        current_minibuf.as_window_or_error().is_live()
+    {
```

Comment:
I think you shouldn't raise errors here.

Reply:
The alternative would be `unwrap()`... not sure that is better.

Reply:
I don't think it's better, but it looks like the c code does not throw errors here.

Reply:
Makes sense. As only windows object can be in `current_minibuf` static var, an error should not occur there. I think we can safely unwrap().